src/transformStateWithClones.js

```diff
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformState = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    let newObj = {};
+
+    switch (action.type) {
+      case 'clear':
+        newObj = { ...clear(cloneState) };
+        break;
+
+      case 'addProperties':
+        newObj = { ...addProperties(cloneState, action.extraData) };
+        break;
+
+      case 'removeProperties':
+        newObj = { ...removeProperties(cloneState, action.keysToRemove) };
```

Comment:
There is no need to spread the result of the functions into a new object since these functions already return a new object. You can assign the result directly to 'cloneState'.